internal/logging/lsp.go

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package logging
+
+import (
+	"context"
+	"log"
+	"strings"
+
+	"github.com/creachadair/jrpc2"
+	lsp "github.com/hashicorp/terraform-ls/internal/protocol"
+)
+
+func NewLspLogger(lspLog *LspLogger) *log.Logger {
+	return log.New(lspLog, "", log.Lshortfile)
+}
+
+type LspLogger struct {
+	Context context.Context
+}
+
+func (l LspLogger) Write(p []byte) (int, error) {
+	logMessage := string(p)
+	// there appears to be an extra newline that's helpful for stderr
+	// but not for outputchannel
+	logMessage = strings.TrimSuffix(logMessage, "\n")
```

Comment:
I wonder if this just originates from our use of some `Println`s in various places? 🤔 

I'd be slightly worried about the performance impact here in such a critical path but maybe it's negligible.